diff --git a/xml-rpc.el b/xml-rpc.el
index 282a5d2..f5714a1 100644
--- a/xml-rpc.el
+++ b/xml-rpc.el
@@ -574,10 +574,9 @@ or nil if called with ASYNC-CALLBACK-FUNCTION."
                                         " encoding=\"UTF-8\"?>\n"
                                         (with-temp-buffer
                                           (xml-print xml)
-                                          (when xml-rpc-allow-unicode-string
-                                            (encode-coding-region
-                                             (point-min) (point-max) 'utf-8))
-                                          (buffer-string))
+                                          (if xml-rpc-allow-unicode-string
+                                              (encode-coding-string (buffer-string) 'utf-8)
+                                            (buffer-string)))
                                         "\n"))
               (url-mime-charset-string "utf-8;q=1, iso-8859-1;q=0.5")
               (url-request-coding-system xml-rpc-use-coding-system)
